src/nifs/protogalaxy/mod.rs

```diff
@@ -76,104 +80,133 @@ impl<C: CurveAffine, const L: usize> ProtoGalaxy<C, L> {
+        let l_0: C::ScalarExt = lagrange_for_gamma
+            .next()
+            .expect("safe, because len of lagrange is `2^log_n`");
+
+        let new_accumulator = RelaxedPlonkWitness {
+            W: acc
+                .W
+                .into_iter()
+                .map(|r| r.into_iter().map(|w| w * l_0).collect())
+                .collect(),
+            E: acc.E.iter().map(|e| *e * l_0).collect(),
+        };
+
+        incoming
+            .zip(lagrange_for_gamma)
+            .fold(new_accumulator, |mut acc, (w, l_n)| {
+                acc.W
+                    .iter_mut()
+                    .zip_eq(w.W.iter())
+                    .for_each(|(acc_W_row, W_row)| {
+                        acc_W_row
+                            .iter_mut()
+                            .zip_eq(W_row.iter())
+                            .for_each(|(acc_cell, cell)| {
+                                *acc_cell += *cell * l_n;
+                            });
+                    });
+
+                acc
+            })
+    }
+
+    fn fold_instance<'i>(
+        acc: RelaxedPlonkInstance<C>,
+        incoming: impl Iterator<Item = &'i PlonkInstance<C>>,
+        mut lagrange_for_gamma: impl Iterator<Item = C::Scalar>,
+    ) -> RelaxedPlonkInstance<C> {
+        let l_0: C::ScalarExt = lagrange_for_gamma
             .next()
             .expect("safe, because len of lagrange is `2^log_n`");
 
         let ecc_mul =
             |pt: C, val: C::ScalarExt| -> C { arithmetic::best_multiexp(&[val], &[pt]).into() };
 
-        let new_accumulator = RelaxedPlonkTrace::<C> {
-            U: RelaxedPlonkInstance {
-                W_commitments: acc
-                    .U
-                    .W_commitments
-                    .into_iter()
-                    .map(|w| ecc_mul(w, l_0))
-                    .collect(),
-                instance: acc.U.instance.into_iter().map(|i| i * l_0).collect(),
-                challenges: acc.U.challenges.into_iter().map(|c| c * l_0).collect(),
-                u: acc.U.u * l_0,
-                // Ignore for protogalaxy
-                E_commitment: C::identity(),
-            },
-            W: RelaxedPlonkWitness {
-                W: acc
-                    .W
-                    .W
-                    .into_iter()
-                    .map(|r| r.into_iter().map(|w| w * l_0).collect())
-                    .collect(),
-                E: acc.W.E.iter().map(|e| *e * l_0).collect(),
-            },
+        let new_accumulator = RelaxedPlonkInstance {
+            W_commitments: acc
+                .W_commitments
+                .into_iter()
+                .map(|w| ecc_mul(w, l_0))
+                .collect(),
+            instance: acc.instance.into_iter().map(|i| i * l_0).collect(),
+            challenges: acc.challenges.into_iter().map(|c| c * l_0).collect(),
+            u: acc.u * l_0,
+            // Ignore for protogalaxy
+            E_commitment: C::identity(),
```

Comment:
it is better to use PlonkInstance as part of Accumulator for protogalaxy without using E (even it is set to default).  Since `E_commitment` is artifact from vanillaFS while `e*, betas` are artifacts from protogalaxyFS.

Reply:
https://github.com/snarkify/sirius/pull/331

Reply:
In next PR